typeform/typeform.py

```diff
@@ -0,0 +1,182 @@
+import time
+import copy
+import json
+import panoply
+import urllib2
+import dateparser
+
+
+FETCH_LIMIT = 500
+DESTINATION = 'typeform'
+DESTINATION_POSTFIX = '_{__table}'
+BASE_URL = 'https://api.typeform.com/v1'
+
+
+class Typeform(panoply.DataSource):
+
+    # constructor
+    def __init__(self, source, options):
+        super(Typeform, self).__init__(source, options)
+
+        if 'destination' not in source:
+            source['destination'] = DESTINATION
+
+        # append the destination postfix, which represent the form name
+        source['destination'] += DESTINATION_POSTFIX
+
+        # configure incremental pulling
+        self._incval = None
+        if (source.get('inckey')):
+            incval = source.get('incval')
+            self._incval = (incval if incval else
+                source.get('lastTimeSucceed') or source.get('lastRuntime'))
+
+            # validate that's the incremental value is a valid
+            # date that could be parsed to a Unix timestamp before
+            # including it in the process
+            self._incval = getTimestamp(self._incval)
+
+
+        forms = source.get('forms')
+        forms = forms if forms else []
```

Comment:
It's a matter of taste but `forms = forms or []`

but you could take both these lines into `forms = source.get('forms') or []`